### ISSUE TITLE
Text wrapping in a table breaks color configuration

### ISSUE THREAD

**Issue by ayala-orca** [jedib0t/go-pretty#330]
**Describe the bug**
When using the table library to format text with by setting `text.WrapText` and `WidthMax`, if the text contains both color and bold formatting, the text wrapping breaks the color configuration. Only the first line is colored as expected.

**To Reproduce**
Run the following code
```
func main() {
	columnHeaders := []interface{}{"Color only", "Color and Bold", "No color"}
	rows := []table.Row{
		{
			color.New(color.FgHiRed).Sprint("This line should be colored in red"),
			color.New(color.FgHiRed).Sprintf("%s %s", color.New(color.Bold).Sprint("Bold Title"), "This line should be colored in red"),
			"This line should be colored in white",
		},
	}

	t := table.NewWriter()
	t.SetOutputMirror(os.Stdout)
	t.Style().Options = table.Options{
		SeparateColumns: true,
		DrawBorder:      true,
	}
	t.AppendHeader(columnHeaders)
	var columnConfig []table.ColumnConfig

	for i := 0; i < len(columnHeaders); i++ {
		columnConfig = append(columnConfig, table.ColumnConfig{Number: i + 1, WidthMaxEnforcer: text.WrapText, WidthMax: 25})
	}
	t.SetColumnConfigs(columnConfig)
	t.AppendRows(rows)
	t.Render()
}
```

**Expected behavior**
The text should retain the color formatting even after being wrapped to the next line.

**Screenshots**
As shown in the second column - the color configuration is broken when the text wrapping occurs.
![image](https://github.com/user-attachments/assets/15e07add-5278-4696-9e05-413044c46b4f)

**Software (please complete the following information):**
 - GoLang Version: 1.22

**Comment by jedib0t** [comment]
What is the `color` library you are using?

**Comment by ayala-orca** [comment]
> What is the `color` library you are using?

https://github.com/fatih/color

**Comment by jedib0t** [comment]
Found the issue, and a workaround.

Issue is because fatih/color library uses `\x1b0;22m` to turn off Bold, and that is not a widely supported instruction, and my text library fails to parse it correctly.

Workaround: Use this:
```golang
			fmt.Sprintf("%s %s",
				color.New(color.FgHiRed, color.Bold).Sprint("Bold Title"),
				color.New(color.FgHiRed).Sprint("This line should be colored in red"),
			),
```
instead of:
```golang
			color.New(color.FgHiRed).Sprintf("%s %s", color.New(color.Bold).Sprint("Bold Title"), "This line should be colored in red"),
```

I'll see what I can do to handle this case without the workaround.

**Comment by jedib0t** [comment]
With the fix in the commit above, the output for your code looks like this: 
![image](https://github.com/user-attachments/assets/75ea8195-1f29-4a4a-b935-26e156001748)

**Comment by jedib0t** [comment]
The fix is part of tag v6.6.0 @ https://github.com/jedib0t/go-pretty/releases/tag/v6.6.0 - please let me know if you face any issues with this release.
